ffi/src/winscard/system_scard/card.rs

```diff
@@ -0,0 +1,403 @@
+                )
+            },
+            "SCardReconnect failed"
+        )?;
+
+        Ok(Protocol::from_bits(active_protocol).unwrap_or_default())
+    }
+
+    fn get_attribute(&self, attribute_id: AttributeId) -> WinScardResult<Cow<[u8]>> {
+        let attr_id = attribute_id
+            .to_u32()
+            .ok_or_else(|| Error::new(ErrorKind::InternalError, "Cannot convert AttributeId -> u32"))?;
+        let mut data_len = 0;
+
+        // https://pcsclite.apdu.fr/api/group__API.html#gaacfec51917255b7a25b94c5104961602
+        // If this value is NULL, SCardGetAttrib() ignores the buffer length supplied in pcbAttrLen, writes the length of the buffer
+        // that would have been returned if this parameter had not been NULL to pcbAttrLen, and returns a success code.
+        //
+        // https://learn.microsoft.com/en-us/windows/win32/api/winscard/nf-winscard-scardgetattrib
+        // If this value is NULL, SCardGetAttrib ignores the buffer length supplied in pcbAttrLen,
+        // writes the length of the buffer that would have been returned if this parameter
+        // had not been NULL to pcbAttrLen, and returns a success code.
+        //
+        // SAFETY: This function is safe to call because `self.h_card` is checked
+        // and other function parameters are type checked.
+        try_execute!(
+            unsafe { (self.api.SCardGetAttrib)(self.h_card()?, attr_id, null_mut(), &mut data_len) },
+            "SCardGetAttrib failed"
+        )?;
+
+        let mut data = vec![0; data_len.try_into()?];
+
+        try_execute!(
+            unsafe { (self.api.SCardGetAttrib)(self.h_card()?, attr_id, data.as_mut_ptr(), &mut data_len) },
+            "SCardGetAttrib failed"
+        )?;
+
+        Ok(Cow::Owned(data))
+    }
+
+    fn set_attribute(&mut self, attribute_id: AttributeId, attribute_data: &[u8]) -> WinScardResult<()> {
+        let attr_id = attribute_id
+            .to_u32()
+            .ok_or_else(|| Error::new(ErrorKind::InternalError, "Cannot convert AttributeId -> u32"))?;
+
+        let len = attribute_data.len().try_into()?;
+
+        // SAFETY: This function is safe to call because `self.h_card` is checked
+        // and other function parameters are type checked.
+        try_execute!(
+            unsafe { (self.api.SCardSetAttrib)(self.h_card()?, attr_id, attribute_data.as_ptr(), len) },
+            "SCardSetAttrib failed"
+        )
+    }
+
+    fn disconnect(&mut self, disposition: ReaderAction) -> WinScardResult<()> {
+        // SAFETY: This function is safe to call because `self.h_card` is always a valid handle
+        // and the `disposition` parameter is type checked.
+        try_execute!(
+            unsafe { (self.api.SCardDisconnect)(self.h_card()?, disposition.into()) },
+            "SCardDisconnect failed"
+        )?;
+
+        // Mark the current card handle as disconnected.
+        self.h_card = HandleState::Disconnected;
+
+        Ok(())
+    }
```

Comment:
praise: Thank you for properly documenting the unsafe blocks!

question: Can you make sure that clippy recognize the comment by adding [this attribute](clippy::undocumented_unsafe_blocks) on the function (no need to commit this specifically):
```rust
#[warn(clippy::undocumented_unsafe_blocks)]
```

Reply:
Cool! I didn't know about this Clippy lint.
done